frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -142,8 +147,17 @@ export const getServerlessData = (resource: K8sResourceKind) => {
 
 export const getDeploymentData = (resource: K8sResourceKind) => {
   let deploymentData = { env: [], replicas: 1, triggers: { image: true, config: true } };
-  if (getResourcesType(resource) !== Resources.KnativeService) {
-    const containers = _.get(resource, 'spec.template.spec.containers', []);
+  const containers = _.get(resource, 'spec.template.spec.containers', []);
+  if (getResourcesType(resource) === Resources.KnativeService) {
+    deploymentData = {
+      ...deploymentData,
+      env: containers[0]?.env || [],
+      triggers: {
+        ...deploymentData.triggers,
```

Comment:
Here we do not need the `config` trigger so you shouldnt be doing `...deploymentData.triggers` as it adds the `config` trigger as well.

Reply:
yes , added as we are using same data here `DeploymentData` and config and replicas are required filed for validation schema so I let the default value flow in. 

To avoid , either i need to make `replicas` and `config` optional in validationSchema here which won't be correct so next alternative is to create own deployment data for serverless. WDYT?

Reply:
I understand why you are doing `...deploymentData`, since `replicas` is required. But you dont need  `...deploymentData.triggers` as the `triggers` from  `...deploymentData` will be overridden by the new  `triggers`.

Reply:
Updated